documentation/tig_stack.md

```diff
@@ -0,0 +1,340 @@
+
+```puppet
+plan tig() {
+
+  apply_prep('all')
+
+  apply('dashboard') {
+    include tig::dashboard
+  }
+
+  $dashboard_host = get_targets('dashboard')[0].name
+
+  apply('agents') {
+    class{ tig::telegraf:
+      influx_host => $dashboard_host
+    }
+  }
+
+  return('grafana_dashboard' => "http://${dashboard_host}:8080")
+}
+```
+
+The first step in the plan uses the `apply_prep` function to install the
+`puppet-agent` package and collect facts from each of the targets in the
+inventory file.
+
+Next, an apply block applies the `dashboard` manifest, which installs and
+configures both Grafana and InfluxDB. The `dashboard` manifest looks like this:
+
+```puppet
+class tig::dashboard (
+  String $grafana_password = $tig::params::grafana_password,
+  String $grafana_user = $tig::params::grafana_user,
+  String $grafana_url = $tig::params::grafana_url,
+  String $influx_password = $tig::params::influxdb_password,
+  String $influx_database = $tig::params::influxdb_database,
+  String $influx_username = $tig::params::influxdb_user,
+
+) inherits ::tig::params {
+  class { 'grafana':
+    cfg => {
+      app_mode => 'production',
+      server   => {
+        http_port     => 8080,
+      },
+      security => {
+        admin_user => $grafana_user,
+        admin_password => $grafana_password,
+      },
+      database => {
+        type          => 'sqlite3',
+        host          => '127.0.0.1:3306',
+        name          => 'grafananana',
+      },
+    },
+  }
+
+  class {'influxdb': }
+  influx_database{$influx_database:
+    superuser => $influx_username,
+    superpass => $influx_password
+  }
+
+  grafana_datasource { 'influxdb':
+    require           => Influx_database['bolt'],
+    grafana_url       => $grafana_url,
+    grafana_user      => $grafana_user,
+    grafana_password  => $grafana_password,
+    type              => 'influxdb',
+    url               => 'http://localhost:8086',
+    user              => $influx_username,
+    password          => $influx_password,
+    database          => $influx_database,
+    access_mode       => 'proxy',
+    is_default        => true,
+  }
+
+  grafana_dashboard { 'telegraf':
+    grafana_url       => $grafana_url,
+    grafana_user      => $grafana_user,
+    grafana_password  => $grafana_password,
+    content           => template('tig/dashboards/telegraf.json')
+  }
+}
+```
+The `dashboard` manifest inherits a
+separate class called `tig::params`, which contains configuration parameters:
+
+```puppet
+class tig::params (
+  String $influxdb_password,
+  String $grafana_password,
+  String $influxdb_database = 'bolt',
+  String $influxdb_user = 'bolt',
+  String $grafana_url = 'http://localhost:8080',
+  String $grafana_user = 'bolt',
+) {}
+```
+
+You can keep the credentials for signing into each service in a separate
+location so they're not part of the manifest code. This implementation stores
+the credentials in `data/common.yaml`:
```

Comment:
I think a note that this uses Hiera, with a link to documentation, would be helpful.

Reply:
It also looks like this file isn't in the tig repo.